### PR TITLE
Beambeam

### PR DESCRIPTION
**Pull request by tkstanczak**

### CODE REVIEW

File: src/Nethermind/Nethermind.Synchronization/FastSync/StateSyncFeed.cs [hunk 1]
```diff
@@ -564,8 +559,8 @@ void AddAgainAllItems()
                         lock (_codeDbLock)
                         {
                             _codeDb[_fastSyncProgressKey.Bytes] = rlp.Bytes;
-                            _codeDb.Commit();
-                            _stateDb.Commit();
+                            // _codeDb.Commit();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Synchronization/FastSync/StateSyncFeed.cs:562]: ?

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitializeNetwork.cs [hunk 2]
```diff
@@ -100,8 +100,15 @@ private async Task Initialize()
             _ctx.SyncPeerPool = new SyncPeerPool(_ctx.BlockTree, _ctx.NodeStatsManager, maxPeersCount, _ctx.LogManager);
             _ctx.DisposeStack.Push(_ctx.SyncPeerPool);
             
-            SyncProgressResolver syncProgressResolver = new SyncProgressResolver(_ctx.BlockTree, _ctx.ReceiptStorage, _ctx.DbProvider.StateDb, _syncConfig, _ctx.LogManager);
+            SyncProgressResolver syncProgressResolver = new SyncProgressResolver(_ctx.BlockTree, _ctx.ReceiptStorage, _ctx.DbProvider.StateDb, _ctx.DbProvider.BeamStateDb, _syncConfig, _ctx.LogManager);
             MultiSyncModeSelector syncModeSelector = new MultiSyncModeSelector(syncProgressResolver, _ctx.SyncPeerPool, _syncConfig, _ctx.LogManager);
+            if (_ctx.SyncModeSelector != null)
+            {
+                // this is really bad and is a result of lack of proper dependency management
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitializeNetwork.cs:107]: maybe just inject delegate to BeaSyncDb in the first place? Or SyncModeSelectorSource ;)

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitializeNetwork.cs:107]: SyncModeSelectorSelector :D

File: src/Nethermind/Nethermind.Synchronization/FastSync/StateSyncFeed.cs (hunk 1 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Synchronization/FastSync/StateSyncFeed.cs:562]: no longer state DB

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Synchronization/FastSync/StateSyncFeed.cs:562]: removed the code that was commented out

**Pull request merged**